cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -155,3 +161,81 @@ def transcript_download_handler(request, course_key_string):
         response = HttpResponseNotFound()
 
     return response
+
+
+def validate_transcript_upload_data(data, files):
+    """
+    Validates video transcript file.
+    Arguments:
+        data: A request's data part.
+        files: A request's files part.
+    Returns:
+        None or String
+        If there is error returns error message otherwise None.
+    """
+    error = None
+    # Validate the must have attributes - this error is unlikely to be faced by common users.
+    must_have_attrs = ['edx_video_id', 'language_code']
+    missing = [attr for attr in must_have_attrs if attr not in data]
+    if missing:
+        error = _(u'Following parameters are required: {missing}.').format(missing=', '.join(missing))
+    elif 'file' not in files:
+        error = _(u'A transcript file is required.')
+
+    return error
+
+
+@login_required
+@require_POST
+def transcript_upload_handler(request, course_key_string):
+    """
+    View to upload a transcript file.
+
+    Arguments:
+        request: A WSGI request object
+        course_key_string: Course key identifying a course
+
+    Transcript file, edx video id and transcript language are required.
+    Transcript file should be in SRT(SubRip) format.
+
+    Returns
+        - A 400 if any of the validation fails
+        - A 404 if the corresponding feature flag is disabled
+        - A 200 if transcript has been uploaded successfully
+    """
+    # Check whether the feature is available for this course.
+    course_key = CourseKey.from_string(course_key_string)
+    if not VideoTranscriptEnabledFlag.feature_enabled(course_key):
+        return HttpResponseNotFound()
+
+    error = validate_transcript_upload_data(data=request.POST, files=request.FILES)
+    if error:
+        response = JsonResponse({'error': error}, status=400)
+    else:
+        edx_video_id = request.POST['edx_video_id']
+        language_code = request.POST['language_code']
+        transcript_file = request.FILES['file']
+        try:
+            # Convert SRT transcript into an SJSON format
```

Comment:
Why are we converting to sjson ?

Reply:
Because we upload to S3 in sjson format. 
this is as video component captions works with sjson transcripts and we don't want to convert from srt to sjson every time on  video component render (if we upload in SRT format).